src/geometry/euclidean/sphere.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import geometry.euclidean.basic
+
+/-!
+# Spheres
+
+This file proves basic geometrical results about distances and angles
+in spheres in real inner product spaces and Euclidean affine spaces.
+-/
+
+open real
+open_locale euclidean_geometry real_inner_product_space real
+
+variables {V : Type*} [inner_product_space ℝ V]
+
+namespace inner_product_geometry
+
+/-!
+### Geometrical results on spheres in real inner product spaces
+
+This section develops some results on spheres in real inner product spaces,
+which are used to deduce corresponding results for Euclidean affine spaces.
+-/
+
+lemma mul_norm_eq_sub_norm_pow_two {x y z : V}
```

Comment:
```suggestion
lemma mul_norm_eq_abs_sub_sq_norm {x y z : V}
```

Reply:
Good suggestion. I think `mul_norm_eq_abs_sub_sq_norm_of_smul_and_norm_eq_norm` is even more descriptive.